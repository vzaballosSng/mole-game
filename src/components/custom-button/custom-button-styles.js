import { css } from 'lit';

export const customButtonStyles = css`
	button {
		background-color: #a0522d;
		border: none;
    border-radius: 10px;
		color: white;
    cursor: pointer;
		padding: 15px 10px;
		text-align: center;
		text-decoration: none;
		font-size: 1rem;
		min-width: 6.25rem;
		height: 3.125rem
	}
`;
