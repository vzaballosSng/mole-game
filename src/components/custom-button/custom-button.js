import { LitElement, html } from 'lit';
import { customButtonStyles } from './custom-button-styles.js';

export class CustomButton extends LitElement {
	static get styles() {
		return [customButtonStyles];
	}
	/**
	 * @property {String} size - Button's size.
	 */
	static get properties() {
		return {
      size: { type: String },
    };
	}

	constructor() {
		super();
    this._buttonClass = {
      'large': 'large',
      'show': 'short'
    }
    this.size = 'short'
	}

	render() {
		return html`
			<button
				class="${this.size}"
				@click="${this._onClick}"
			>
			  <slot></slot>
			</button>
		`;
	}
}

window.customElements.define('custom-button', CustomButton);
