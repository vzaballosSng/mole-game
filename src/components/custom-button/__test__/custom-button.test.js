import { html, fixture } from '@open-wc/testing';
import { getAllElementsAndShadowRoots } from 'shadow-dom-testing-library';
import { expect } from '@jest/globals';
import '../custom-button.js';

describe('custom-button', () => {
	it('matches the snapshot', async () => {
		const el = await fixture(html`<custom-button></custom-button>`);
		const data = getAllElementsAndShadowRoots(el, '*');
		expect(data).toMatchSnapshot();
	});

	it('matches the snapshot slot', async () => {
		const el = await fixture(
			html`<custom-button></custom-button>`
		);
		const data = getAllElementsAndShadowRoots(el, '*');
		expect(data).toMatchSnapshot();
	});
});
