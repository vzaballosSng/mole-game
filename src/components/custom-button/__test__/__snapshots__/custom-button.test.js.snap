// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-button matches the snapshot 1`] = `
[
  <custom-button />,
  ShadowRoot {},
  <button
    class="short"
  >
    
			  
    <slot />
    
			
  </button>,
  <slot />,
  <style>
    
	button {
		background-color: #a0522d;
		border: none;
    border-radius: 10px;
		color: white;
    cursor: pointer;
		padding: 15px 10px;
		text-align: center;
		text-decoration: none;
		font-size: 1rem;
		min-width: 6.25rem;
		height: 3.125rem
	}

  </style>,
]
`;

exports[`custom-button matches the snapshot slot 1`] = `
[
  <custom-button />,
  ShadowRoot {},
  <button
    class="short"
  >
    
			  
    <slot />
    
			
  </button>,
  <slot />,
  <style>
    
	button {
		background-color: #a0522d;
		border: none;
    border-radius: 10px;
		color: white;
    cursor: pointer;
		padding: 15px 10px;
		text-align: center;
		text-decoration: none;
		font-size: 1rem;
		min-width: 6.25rem;
		height: 3.125rem
	}

  </style>,
]
`;
